### ISSUE TITLE
coreutils shred-test fails on bcachefs filesystem

### ISSUE THREAD

**Issue by JohnRTitor** [koverstreet/bcachefs-tools#270]
As reported in https://github.com/NixOS/nixpkgs/issues/306760 and https://github.com/NixOS/nixpkgs/issues/313700
[`shred-exact` test](https://github.com/coreutils/coreutils/blob/master/tests/shred/shred-exact.sh) of coreutils is failing on `bcachefs` filesystem.

Relevant log:
```
FAIL: tests/shred/shred-exact
=============================

shred: file.slop: error writing at offset 1048576
FAIL tests/shred/shred-exact.sh (exit status: 1)
```

CC: @jcaesar @Jayman2000

**Comment by koverstreet** [comment]
I'll need a real log, or something to go off of...

**Comment by koverstreet** [comment]
strace shows that it's doing a misaligned DIO write - we were failing to return an error in that case.

Fixed by a316668abbf2
